subprojects/internal-performance-testing/src/main/groovy/org/gradle/performance/generator/FileContentGenerator.groovy

```diff
@@ -0,0 +1,199 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.performance.generator
+
+class FileContentGenerator {
+
+    TestProjectGeneratorConfiguration config
+
+    FileContentGenerator(TestProjectGeneratorConfiguration config) {
+        this.config = config
+    }
+
+    def generateBuildGradle(Integer subProjectNumber, DependencyTree dependencyTree) {
+        def isRoot = subProjectNumber == null
+        if (isRoot && config.subProjects > 0) {
+            return ""
+        }
+        """
+        ${config.plugins.collect { "apply plugin: '$it'" }.join("\n        ") }
+        
+        String compilerMemory = getProperty('compilerMemory')
+        String testRunnerMemory = getProperty('testRunnerMemory')
+        int testForkEvery = getProperty('testForkEvery') as Integer
+
+        repositories {
+            ${config.repositories.join("\n            ")}
+        }      
+        
+        configurations {
+            compile.extendsFrom projectsConfiguration
+        }
+
+        dependencies {
+            ${config.externalApiDependencies.collect { "api '$it'" }.join("\n            ")}
+            ${config.externalImplementationDependencies.collect { "implementation '$it'" }.join("\n            ")}
+            testImplementation '${config.useTestNG ? 'org.testng:testng:6.4' : 'junit:junit:4.12'}'
+
+            ${subProjectNumber == null ? '' : dependencyTree.parentToChildrenNodeSets.get(subProjectNumber).collect { "${it % 9 == 0 ? 'api' : 'implementation'} project(':project$it')" }.join("\n            ")}
```

Comment:
implementation/api only work with >3.4. We need to use `compile` for older versions.